File: src/laser2pc.cpp
#include <ros/ros.h>
#include <sensor_msgs/PointCloud2.h>
#include <sensor_msgs/LaserScan.h>
#include <laser_geometry/laser_geometry.h>

class laser2pc
{
public:
    laser2pc();

private:
    ros::NodeHandle nh;
    laser_geometry::LaserProjection projector;
    ros::Subscriber laser_sub;
    ros::Publisher pc_pub;

    void laser_callback(const sensor_msgs::LaserScan::ConstPtr& laser_message);

};

laser2pc::laser2pc()
{
    laser_sub = nh.subscribe("scan", 10, &laser2pc::laser_callback, this);
    pc_pub = nh.advertise<sensor_msgs::PointCloud2>("laser2pc/pc2", 1);
}

void laser2pc::laser_callback(const sensor_msgs::LaserScan::ConstPtr& laser_message)
{
    sensor_msgs::PointCloud2 cloud;

    //transform LaserScan -> PointCloud2
    projector.projectLaser(*laser_message, cloud);

    pc_pub.publish(cloud);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "laser2pc");
    laser2pc l2p;

    ros::spin();
    return 0;
}

File: scripts/laser2pc.py
#!/usr/bin/python -t
# -*- coding: utf-8 -*-

"""
@brief 
@author Toshihiko Shimizu
@updated 2019/09/17 22:0:46 @
@ref     https://www.youtube.com/watch?v=GvilxcePD64
"""
import rospy
from sensor_msgs.msg import PointCloud2 as pc2
from sensor_msgs.msg import LaserScan
from laser_geometry import LaserProjection

class Laser2PC():
    def __init__(self):
        self.laserProj = LaserProjection()
        self.pcPub = rospy.Publisher("laser2pc/pc2", pc2, queue_size=1)
        self.laserSub = rospy.Subscriber("scan", LaserScan, self.laserCallback)

    def laserCallback(self, data):
        cloud_out = self.laserProj.projectLaser(data)
        self.pcPub.publish(cloud_out)


if __name__ == '__main__':
    rospy.init_node("laser2PointCloud")
    l2pc = Laser2PC()
    rospy.spin()